src/convertToObject.js

```diff
@@ -12,7 +12,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssPairs = sourceString.split('\n').join('').split(';');
+  // morfing cssPairs string by removing \n
+  // and splitting it by ';' to key/value strings
+  const cssParams = {}; // our resulting object
+
+  for (let item of cssPairs) {
+    item = item.split(':'); // spliting to key/value pairs
+    item[0] = item[0].trim();
+
+    if (item[0].length === 0) { // check if key is invalid
+      continue; // meaning pair will be invalid as well
+    }
+    item[1] = item[1].trim(); // if pair valid triming value
+    cssParams[item[0]] = item[1]; // assigning key/value to object
+  }
+
+  return cssParams;
```

Comment:
```suggestion
  const cssPairs = sourceString
  .split('\n').join('')
  .split(';')
  .map((rule) => rule.split(':'))
  .filter([key, value] => key.trim() && value.trim());
  
  const cssParams = {};

  for (let [key, value] of cssPairs) {
    cssParams[key] = value;
  }

  return cssParams;
```

map нужен для чего-то такого, ты сразу все значения проверяешь через трим, и фильтруешь, в итоге в массиве у тебя остаются только валидные значения. И в цикле нужно их все просто добавлять, без каких-либо проверок

Reply:
Спасибо! Я ещё добавил trim на момент добавления в объект, так как у меня получилось что при проверке валидности значения подрезаются, но только чтоб валидность проверить, а в строках/парах они остаются необрезанными.